itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/create/CreateApplicationOnBuilderImageTest.java

```diff
@@ -190,11 +193,21 @@ public void tearDown() {
 		explorer.reopen();
 
 		OpenShift3Connection connection = explorer.getOpenShift3Connection(connectionReq.getConnection());
-		connection.getProject().delete();
+		connection.refresh();
+		connection.getProject(DatastoreOS3.PROJECT1_DISPLAYED_NAME).delete();
 
 		ProjectExplorer projectExplorer = new ProjectExplorer();
 		if (projectExplorer.containsProject(projectName)) {
 			projectExplorer.getProject(projectName).delete();
 		}
+		
+		connection.refresh();
+		
+		//Close all shells if test fails (could interfere next tests)
+		WorkbenchShellHandler.getInstance().closeAllNonWorbenchShells();
```

Comment:
Just curious, this will close all opened/left shells like error dialogs, wizards, etc?

Reply:
Yes, it should close all, in case the test fails.

Reply:
Classy, do you mind if I use it as well? :)